src/Components/Components/src/PersistentState/PersistentServiceTypeCache.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Since the cache is identical with `RootComponentTypeCache`, what's the motivation to have it as a copy?

Reply:
I wanted to avoid generalizing and renaming the type, as it creates an explosion on the number of files to touch in the PR.

I'm doing it, but I don't think there's any value in avoiding duplication here.